go_to_office/lib/screens/admin/offices_list.dart

```diff
@@ -1,69 +1,89 @@
-              child: Text(Strings.add_new_office),
-              onPressed: () {
-                Navigator.push(
-                  context,
-                  MaterialPageRoute(builder: (context) => OfficePage(null)),
-                );
-              },
-            ),
-            ListView.builder(
-                scrollDirection: Axis.vertical,
-                shrinkWrap: true,
-                padding: const EdgeInsets.all(8),
-                itemCount: officesList.length,
-                itemBuilder: (BuildContext context, int index) {
-                  return new GestureDetector(
-                    onTap: () => openOfficePage(officesList[index].id),
-                    child: Container(
-                        height: 50,
-                        margin: EdgeInsets.all(2),
-                        child: Text(
-                          '${officesList[index].name}',
-                          style: TextStyle(fontSize: 18),
-                        )),
-                  );
-                })
-          ]),
-        ));
+    return FutureBuilder(
+        future: getOfficesListApi(),
+        builder: (context, snapshot) {
+          if (snapshot.connectionState == ConnectionState.done &&
+              snapshot.hasData) {
+            Text('we have data}');
+            var officesListLocal = snapshot.data;
+            return Scaffold(
+                appBar: AppBar(
+                  title: Text(Strings.offices),
+                ),
+                body: SingleChildScrollView(
+                  padding: EdgeInsets.all(15.0),
+                  child:
+                      Column(mainAxisSize: MainAxisSize.min, children: <Widget>[
+                    ElevatedButton(
+                      child: Text(Strings.add_new_office),
+                      onPressed: () {
+                        openOfficePage(Office(), context);
+                      },
+                    ),
+                    ListView.builder(
+                        scrollDirection: Axis.vertical,
+                        shrinkWrap: true,
+                        padding: const EdgeInsets.all(8),
+                        itemCount: officesListLocal.length,
+                        itemBuilder: (BuildContext context, int index) {
+                          return new GestureDetector(
+                            onTap: () => openOfficePage(
+                                officesListLocal[index], context),
+                            child: Container(
+                                height: 50,
+                                margin: EdgeInsets.all(2),
+                                child: Text(
+                                  '${officesListLocal[index].name}',
+                                  style: TextStyle(fontSize: 18),
+                                )),
+                          );
+                        })
+                  ]),
+                ));
+          } else {
+            return Transform.scale(
```

Comment:
Let's add an debug print for when we don't get the data

Reply:
The solution of showing a spinner - looks like a solution that is used in code examples